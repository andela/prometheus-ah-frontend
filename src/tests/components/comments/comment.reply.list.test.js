import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedCommentReplyList,
{ CommentReplyList } from '../../../components/comment/CommentReplyList';
import mockData from '../../__mocks__/commentMockData';

const props = {
  createCommentReplyAction: jest.fn(),
  deleteCommentAction: jest.fn(),
  updateCommentAction: jest.fn(),
  loadCommentReplyAction: jest.fn(),
  loadCommentsAction: jest.fn(),
  article: mockData.articles[0],
  comment: mockData.comments.comments[0],
  auth: true,
  error: {},
  username: 'faksam',
  commentReply: {
    id: 2,
    createdAt: '2016-02-18T03:22:56.637Z',
    updatedAt: '2016-02-18T03:48:35.824Z',
    body: 'This article is hot',
    User: {
      username: 'joeeasy',
      email: 'joeeasy@gmail.com',
      bio: 'tomorrow is a better day',
      id: 1,
      isVerified: true,
      role: 'user'
    }
  },
  commentReplies: [
    {
      replies: [{
        id: 2,
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        body: 'This article is hot',
        User: {
          username: 'joeeasy',
          email: 'joeeasy@gmail.com',
          bio: 'tomorrow is a better day',
          id: 1,
          isVerified: true,
          role: 'user'
        }
      },
      {
        id: 3,
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        body: 'This article is hot',
        User: {
          username: 'joeeasy',
          email: 'joeeasy@gmail.com',
          bio: 'tomorrow is a better day',
          id: 1,
          isVerified: true,
          role: 'user'
        }
      },
      {
        id: 4,
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        body: 'This article is hot',
        User: {
          username: 'joeeasy',
          email: 'joeeasy@gmail.com',
          bio: 'tomorrow is a better day',
          id: 1,
          isVerified: true,
          role: 'user'
        }
      }],
      commentId: 1
    }, {
      replies: [{
        id: 2,
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        body: 'This article is hot',
        User: {
          username: 'joeeasy',
          email: 'joeeasy@gmail.com',
          bio: 'tomorrow is a better day',
          id: 1,
          isVerified: true,
          role: 'user'
        }
      },
      {
        id: 3,
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        body: 'This article is hot',
        User: {
          username: 'joeeasy',
          email: 'joeeasy@gmail.com',
          bio: 'tomorrow is a better day',
          id: 1,
          isVerified: true,
          role: 'user'
        }
      },
      {
        id: 4,
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        body: 'This article is hot',
        User: {
          username: 'joeeasy',
          email: 'joeeasy@gmail.com',
          bio: 'tomorrow is a better day',
          id: 1,
          isVerified: true,
          role: 'user'
        }
      }],
      commentId: 1
    }],
};
const initialState = {
  articleReducer: {
    article: mockData.articles[0],
  },
  auth: {
    error: {}
  },
  commentReducer: {
    commentReply: mockData.commentReplies[0],
    commentReplies: props.commentReplies,
  },

};
const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);

describe('CommentReplyList Component', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<CommentReplyList {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set state to props', () => {
    const wrapper = shallow(<ConnectedCommentReplyList store={store} {...props} />);
    expect(wrapper.state()).toEqual({});
  });
});
