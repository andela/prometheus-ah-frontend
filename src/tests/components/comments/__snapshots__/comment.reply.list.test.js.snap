// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentReplyList Component should render without throwing an error 1`] = `
<div>
  <div>
    <Connect(CommentReplyRow)
      comment={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 1,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      commentReply={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 2,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      key="2"
    />
    <Connect(CommentReplyRow)
      comment={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 1,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      commentReply={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 3,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      key="3"
    />
    <Connect(CommentReplyRow)
      comment={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 1,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      commentReply={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 4,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      key="4"
    />
    <Connect(CommentReplyRow)
      comment={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 1,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      commentReply={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 2,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      key="2"
    />
    <Connect(CommentReplyRow)
      comment={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 1,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      commentReply={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 3,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      key="3"
    />
    <Connect(CommentReplyRow)
      comment={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 1,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      commentReply={
        Object {
          "User": Object {
            "bio": "tomorrow is a better day",
            "email": "joeeasy@gmail.com",
            "id": 1,
            "isVerified": true,
            "role": "user",
            "username": "joeeasy",
          },
          "body": "This article is hot",
          "createdAt": "2016-02-18T03:22:56.637Z",
          "id": 4,
          "updatedAt": "2016-02-18T03:48:35.824Z",
        }
      }
      key="4"
    />
  </div>
</div>
`;
